datasources/telegram/models.go

```diff
@@ -0,0 +1,243 @@
+	Name     string            `json:"name"`
+	ChatType string            `json:"type"`
+	Messages []telegramMessage `json:"messages"`
+
+	ownerID          string
+	ownerName        string
+	firstMessageTime time.Time
+}
+
+func (item telegramChat) Timestamp() time.Time {
+	return item.firstMessageTime
+}
+
+func (item telegramChat) Owner() (id *string, name *string) {
+	return &item.ownerID, &item.ownerName
+}
+
+func (item telegramChat) DataText() (*string, error) {
+	return nil, nil
+}
+
+func (item telegramChat) DataFileName() *string {
+	return nil
+}
+
+func (item telegramChat) DataFileReader() (io.ReadCloser, error) {
+	return nil, nil
+}
+
+func (item telegramChat) DataFileHash() []byte {
+	return nil
+}
+
+func (item telegramChat) DataFileMIMEType() *string {
+	return nil
+}
+
+func (item telegramChat) Metadata() (*timeliner.Metadata, error) {
+	return nil, nil
+}
+
+func (item telegramChat) Location() (*timeliner.Location, error) {
+	return nil, nil
+}
+
+//  ------------------------------- Telegram Message ---------------------------------------------------------
+
+type telegramMessage struct {
+	MessageID           int                         `json:"id"`
+	Date                string                      `json:"date"`
+	Edited              string                      `json:"edited"`
+	From_id             int                         `json:"from_id"`
+	From                string                      `json:"from"`
+	Text                telegramMessageText         `json:"text,omitempty"`
+	MediaType           string                      `json:"media_type,omitempty"`
+	FileRaw             string                      `json:"file,omitempty"`
+	Thumbnail           string                      `json:"thumbnail,omitempty"`
+	Width               int                         `json:"width,omitempty"`
+	Height              int                         `json:"height,omitempty"`
+	PhotoRaw            string                      `json:"photo,omitempty"`
+	MimeType            string                      `json:"mime_type,omitempty"`
+	ViaBot              string                      `json:"via_bot,omitempty"`
+	DurationSeconds     int                         `json:"duration_seconds,omitempty"`
+	LocationInformation telegramLocationInformation `json:"location_information,omitempty"`
+
+	AbsFilePath    string
+	FromIDParsed   string
+	DateParsed     time.Time
+	EditedParsed   time.Time
+	ConversationID string
+}
+
+type telegramComplexMessageContent struct {
+	Type string `json:"type"`
+	Text string `json:"text"`
+}
+
+type telegramLocationInformation struct {
+	Latitude  float64 `json:"latitude"`
+	Longitude float64 `json:"longitude"`
+}
+
+type telegramMessageText string
+
+// the text attribute is sometimes just a string and sometimes an array of strings or objects, that themselves contain a text and type attribute
+// this type either returns the string or concatenates all strings and object's text attributes into a string
```

Comment:
Shorter comment lines would be nice.

Also, follow godoc convention: "UnmarshalJSON ..."